### DISCUSSION TITLE
Driving cycle interpolant in experiment class

### DISCUSSION THREAD

**Discussion by dion-w** [pybamm-team/PyBaMM#1904] [Q&A] [answered]
Hello everyone, 
I noticed a, for me, strange behaviour in pybamm. It says you can use driving cycles in the experiment class but in other posts I saw that you typically need to interpolate your driving cycle first before using it. However the interpolant doesn't work when being used in the experiment class as pybamm says it can't index into it. 
What is the recommended treatment of this problem or would you consider it an issue? How does it work, meaning how does pybamm handle it if you use a non-interpolated cycle in the experiment class?

**Comment by rtimms** [comment, marked as answer]
Hi, when using a drive cycle in an experiment you don't need to create the interpolant first (the simulation creates it for you). See [this example](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/scripts/experiment_drive_cycle.py). I appreciate this is a bit inconsistent, and maybe we should set this up differently?
